src/js/components/legend/Legend.tsx

```diff
@@ -50,6 +59,24 @@ const Legend = (): JSX.Element => {
   const [visibleLayersToShow, setVisibleLayersToShow] = useState<LayerProps[]>(
     []
   );
+
+  useEffect(() => {
+    const handleResize = () => {
+      setWindowDimensions(getWindowDimensions());
+    };
+
+    window.addEventListener('resize', handleResize);
+    window.removeEventListener('resize', handleResize);
```

Comment:
This snippet adds event listener and removes it immediately? I am assuming we want the event listener for the duration of the component lifecycle? If so, we need to return the clean up function per React Docs: https://reactjs.org/docs/hooks-effect.html

```suggestion
    () => window.removeEventListener('resize', handleResize);
```